### PR TITLE
[BUILD] Setup CircleCI

### PR DESCRIPTION
**Pull request by KamiKillertO**

### CODE REVIEW

**Comment by KamiKillertO** [comment]
@xcambar or @GreatWizard can one of you enable circleci for this project ?

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  version: 2
+  default:
+    jobs:
+      - checkout_code
+      - install_dependencies:
+          requires:
+            - checkout_code
+      - tests:
+          requires:
+            - install_dependencies
```
Comment (ryuran) [inline .circleci/config.yml]: end of file blank line missing
we should add an editor config in this project

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+          at: .
+      - restore_cache:
+          keys:
+            - styledown-node10-v1-{{ checksum "yarn.lock" }}
+      - run:
+          name: Avoid hosts unknown for github
+          command: mkdir ~/.ssh/ && echo -e "Host github.com\n\tStrictHostKeyChecking no\n" > ~/.ssh/config
+      - run:
+          name: Yarn Install
+          command: yarn install --non-interactive --pure-lockfile
```
Comment (ryuran) [inline .circleci/config.yml]: Currently this project do not use yarn but npm with `package-lock.json`

File: .circleci/config.yml (hunk 1 above)
Comment (KamiKillertO) [reply .circleci/config.yml]: Yes, the project also need to be linted.
Feel free to open a PR for the editor config

**Pull request merged**